### DISCUSSION TITLE
Why does Vercel return status code 200 for edge cache hits?

### DISCUSSION THREAD

**Discussion by zoul** [vercel/community#624] [Help]
I’m trying to wrap my head around edge caching in Vercel. We have a serverless function that adds a caching header:

```javascript
response.setHeader("Cache-Control", "max-age=86400, s-maxage=86400");
```

In our logs we see a lot of 200s accessing this endpoint – which is surprising, because we would expect most of them to be 304 / Not Modified. Trying with `curl` I get this:

```bash
% curl --head 'https://…'
HTTP/2 200 
content-type: image/jpeg
x-matched-path: /api/resize
cache-control: max-age=86400
date: Wed, 25 May 2022 14:49:23 GMT
etag: "2da26-CWG3/fz+7RLMQCBlv3q/uSOdxNc"
content-length: 0
x-vercel-cache: HIT
age: 318
server: Vercel
x-vercel-id: fra1::sfo1::mcgdc-1653490481926-2607067a3755
strict-transport-security: max-age=63072000
```

So in fact the edge cache seems to be a hit (`x-vercel-cache: HIT`), but still the status code is 200. According to the [documentation](https://vercel.com/docs/concepts/edge-network/x-vercel-cache) I would expect cache hits to be served with the 304 status code. What am I missing? Thank you!

**Comment by gbertb** [comment]
@zoul Did you ever figure this out? I'm having the same issues

**Reply by zoul** [reply]
No, unfortunately not.
